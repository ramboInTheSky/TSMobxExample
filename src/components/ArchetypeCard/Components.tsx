import styled from '@emotion/styled'
import { Paper, IconButton } from '@material-ui/core'

import { variables } from '../../constants/theme'

export const Wrapper: React.ComponentType<any> = styled(Paper)`
    margin-top: ${variables.gutter};
    background-color: ${variables.componentBackgroundClear} !important;
`
export const Header = styled.div`
    background-color: ${variables.archetypeColor};
    color: #333;
    font-size: 1.3rem;
    height: 25px;
    padding: ${variables.indentation};
    display: flex;
    justify-content: space-between;
`

export const Content: React.ComponentType<any> = styled.div`
    background-color: ${variables.componentBackgroundClear} !important;
    padding: ${variables.indentation};
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
`

export const CloseButton: React.ComponentType<any> = styled(IconButton)`
        padding: 2px !important;
`

export const LockGroup: React.ComponentType<any> = styled(Paper)`
    background-color: ${variables.componentBackground} !important;
    margin: ${variables.indentation};
    padding: ${variables.indentation};
    padding-right: calc(${variables.indentation} + 6px);
`