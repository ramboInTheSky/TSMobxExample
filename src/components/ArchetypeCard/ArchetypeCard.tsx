import React from 'react'
import MeetingRoomIcon from '@material-ui/icons/MeetingRoom'
import CloseIcon from '@material-ui/icons/Close'
import { Wrapper, Header, Content, LockGroup, CloseButton } from './Components'
import { Archetype } from '../../models/Visit';

export interface ArchetypeCardProps {
    archetype?: Archetype
    removeItemFn?: Function
}

export const ArchetypeCard = (props: ArchetypeCardProps) => {
    const { archetype, removeItemFn } = props
    return archetype ? <Wrapper>
        <Header>
            <span>{archetype.name}</span>
            {removeItemFn && <CloseButton onClick={(e: any) => removeItemFn!(e, archetype.id)}><CloseIcon /></CloseButton>}
        </Header>
        <Content>
            {archetype.lockGroupNames.map(item =>
                <LockGroup key={item}>
                    <MeetingRoomIcon /> {item}
                </LockGroup>)}
        </Content>
    </Wrapper> : null
}
