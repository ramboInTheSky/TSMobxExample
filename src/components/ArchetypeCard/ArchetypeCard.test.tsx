import React from 'react'
import { shallow } from 'enzyme'

import { ArchetypeCard } from './ArchetypeCard'

describe('<ArchetypeCard />', () => {
	let props: any
	beforeEach(()=>{
		props = {
			
		}
	})
	it('renders correctly', () => {
		expect(shallow(<ArchetypeCard {...props}/>)).toMatchSnapshot()
	})

	// it('passes on props correctly', () => {
	// 	expect(shallow(<ArchetypeCard foo="bar">Hello</ArchetypeCard>)).toMatchSnapshot()
	// })

	// it('Does not re-render on children props change', () => {
	// 	const el: any = shallow(<ArchetypeCard foo="bar" />)
	// 	el.setProps({ children: 'Goodbye' })
	// 	el.update()
	// 	expect(el).toMatchSnapshot()
	// })
})
