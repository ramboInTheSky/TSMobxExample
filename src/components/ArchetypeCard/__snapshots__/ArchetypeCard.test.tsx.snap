// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArchetypeCard /> renders correctly 1`] = `""`;
